extensions-core/s3-extensions/src/test/java/org/apache/druid/storage/s3/output/RetryableS3OutputStreamTest.java

```diff
@@ -284,13 +304,15 @@ private void assertCompleted(long chunkSize, long expectedFileSize)
       Set<Integer> partNumbersFromRequest = partRequests.stream().map(UploadPartRequest::getPartNumber).collect(Collectors.toSet());
       Assert.assertEquals(partRequests.size(), partNumbersFromRequest.size());
 
-      for (int i = 0; i < partRequests.size(); i++) {
-        if (i < partRequests.size() - 1) {
-          Assert.assertEquals(chunkSize, partRequests.get(i).getPartSize());
-        } else {
-          Assert.assertTrue(chunkSize >= partRequests.get(i).getPartSize());
-        }
-      }
+      // Validate number of chunks uploaded.
+      final long numChunksOfChunkSize = expectedFileSize / chunkSize;
+      final int numChunksOfSmallerSize = expectedFileSize % chunkSize == 0 ? 0 : 1;
+      final long numOfExactChunks = partRequests.stream().filter(part -> part.getPartSize() == chunkSize).count();
+      final long numOfSmallerChunks = partRequests.stream().filter(part -> part.getPartSize() < chunkSize).count();
+
+      Assert.assertEquals(numChunksOfChunkSize, numOfExactChunks);
+      Assert.assertEquals(numChunksOfSmallerSize, numOfSmallerChunks);
```

Comment:
Rephrase for readability:
```suggestion
      
      // Verify sizes of uploaded chunks
      int numSmallerChunks = 0;
      for (UploadPartRequest part : partRequests) {
          Assert.assertTrue(part.getPartSize() <= chunkSize);
          if (part.getPartSize() < chunkSize) {
              ++numSmallerChunks;
          }
      }
      Assert.assertTrue(numSmallerChunks <= 1)
```

Reply:
Done.